src/main/java/com/example/alaa/university/service/JwtService.java

```diff
@@ -0,0 +1,70 @@
+package com.example.alaa.university.service;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.io.Decoders;
+import io.jsonwebtoken.security.Keys;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Service;
+
+import java.security.Key;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.function.Function;
+
+@Service
+public class JwtService implements IJwtService {
+    private static final String SECRET_KEY = "b66acd0c8d99a1283bc29f300d10e405c8a56d57e114a67bfabba7ae35fd551d";
```

Comment:
if you want to have a secret as well.  For better approach it is better to put in in application properties and then read it .  You can add a property in application.properties like
application.secretKey=yourkeyxxx1231231

Than add in counstructor argument - @Value("${application.secretKey}") String secretKet and assing that value to a dynamic field

This change is optional